lxd/instance/instance_utils.go

```diff
@@ -140,6 +175,18 @@ func ValidConfig(sysOS *sys.OS, config map[string]string, profile bool, expanded
 			return fmt.Errorf("Image keys can only be set on instances")
 		}
 
+		if !profile {
+			if instanceType == instancetype.VM {
```

Comment:
Lets simplify this bit based on the change suggested in https://github.com/lxc/lxd/pull/8970/files#r662828712

You can just do:

```go
for specificInstType, specificKeys := range specificInstTypeConfigKeys {
    if specificInstType == instanceType {
        continue
    }

    if shared.StringInSlice(k, specificKeys) {
         return fmt.Errorf("Config key %q is not allowed for instance type %s", k, instanceType)
    }
}

```

Reply:
Also in `isContainerSpecificConfigKey` I'm not sure using ` strings.Contains` is a good idea as it may mismatch partial string keys. Was this intentionally added? Rather than using exact match as in my example above?

Reply:
i used strings.Contains because some key namespaces are instance type exclusive like "limits.hugepage" for Containers, and i could check on the whole namespace instead of key.

Reply:
@my5t3ry ah I see so you're intention was to do a prefix check. Using `strings.Contains` is still not desirable because it may match a future part of a key unintentionally. `strings.HasPrefix()` would be better then in that case. Although we would need to define a different validation function for each key in the list because some should be exact match and not allow prefixes, and others are prefix based. Although in this case limits.hugepages.* are all well defined keys and so shouldn't be using prefix match anyway.